### PR TITLE
Updated database session code to avoid duplication of session id

### PR DESCRIPTION
**Pull request by ghost**
See https://github.com/joomla/joomla-platform/pull/454 for original request.  Repository reorg removed the old fix, so here is a new, updated one.

### CODE REVIEW

**Comment by piotr-cz** [comment]
@garyamort
Please correct the checkstyle errors: tabs for indent and comments starting with capital, see http://developer.joomla.org/pulls/pulls/1209.html for details

We really need this fixed :)
Without this Joomla crashes on multiple concurrent requests (ajax/ thumbnails for image galleries etc).

I've added this fix as a separate Storage storage handler 5 months ago to my app and didn't notice any new issues.

**Comment by elinw** [comment]
http://developer.joomla.org/pulls/pulls/1209.html

libraries/joomla/session/storage/database.php:116
Please consider the /\* */ style for comments that span over multiple lines.
libraries/joomla/session/storage/database.php:117
Whitespace found at end of line
libraries/joomla/session/storage/database.php:119
Tabs must be used to indent lines; spaces are not allowed

File: libraries/joomla/session/storage/database.php [hunk 1]
```diff
@@ -113,22 +113,10 @@ public function write($id, $data)
...
-				$query->insert($db->quoteName('#__session'))
-				->columns($db->quoteName('session_id') . ', ' . $db->quoteName('data') . ', ' . $db->quoteName('time'))
-				->values($db->quote($id) . ', ' . $db->quote($data) . ', ' . $db->quote((int) time()));
-
-				// If the session does not exist, we need to insert the session.
-				$db->setQuery($query);
-				return (boolean) $db->execute();
-			}
+			// Since $db->execute did not throw an exception, so the query was successful.
+			// Either the data changed, or the data was identical.
```
Comment (elinw) [inline libraries/joomla/session/storage/database.php]: I think you could make these two lines and make Jenkins happy.

**Comment by realityking** [comment]
Still 3 code style issues: http://developer.joomla.org/pulls/pulls/1209.html

**Comment by LouisLandry** [comment]
Thanks Gary!

**Pull request merged**